python_tests/test_python_frontend.py

```diff
@@ -70,8 +86,12 @@ def inner_fn(*args, **kwargs):
         self, fusion_func, inputs = args
 
         # NOTE: For debug purposes, clear FusionCache before running first test
-        # if ("new_fusion_expected" not in kwargs) or kwargs["new_fusion_expected"]:
-        #    FusionCache.reset()
+        is_new_fusion_expected = ("new_fusion_expected" not in kwargs) or kwargs[
+            "new_fusion_expected"
+        ]
+        if debug_serde and is_new_fusion_expected:
```

Comment:
For my education, why `is_new_fusion_expected` imply a cache reset? From the naming, they seem to be orthogonal.

Reply:
When `is_new_fusion_expected` is True, the test expects a new fusion to be added to the cache. Otherwise, an existing fusion is reused.

For serialization, we're allowed to reset the cache only when `is_new_fusion_expected` is True.